frontends/web/src/routes/device/bitbox02/setup/wallet-restore.tsx

```diff
@@ -72,22 +72,34 @@ export const RestoreFromMnemonic = ({
   onAbort,
 }: Props) => {
   const { t } = useTranslation();
+  const [errorCode, setErrorCode] = useState<number>();
+  const inProgress = useRef<string>();
 
   useEffect(() => {
-    restoreFromMnemonic(deviceID)
-      .then(result => {
-        if (!result.success) {
-          const errorText = result.code === errUserAbort
-            ? t('bitbox02Wizard.restoreFromMnemonic.e104')
-            : t('bitbox02Wizard.restoreFromMnemonic.failed');
-          alertUser(errorText, {
-            asDialog: false,
-            callback: () => onAbort(),
-          });
-        }
-      })
-      .catch(console.error);
-  }, [deviceID, onAbort, t]);
+    if (inProgress.current !== deviceID) {
+      // remember for which device the workflow already has been initialized
+      inProgress.current = deviceID;
+      restoreFromMnemonic(deviceID)
+        .then(result => {
+          if (!result.success) {
+            setErrorCode(result.code);
```

Comment:
don't you have to reset inProgress.current? Maybe if you abort or an error happens, you cannot start it again?

Reply:
There is no way to start the logic from within this component.

The workflow starts right-aways when this View is mounted.

When the workflow ends it will either load a different view (controlled in parent wizard logic) or show an error + onAbort callback that calls a function from wizard.

does my explanation make sense more or less?